### ISSUE TITLE
Add aria-label to the Slide-Panel container

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#889]
# Bug Report

## Description
UX has indicated that when the slide panel opens, it should focus on the panel, not the first interact-able element.

@scottwilmarth mentioned:

> Focus should go to the panel after a user selects an option that opens a detail panel. Having focus go to the panel allows the assistive technology user (screen reader for instance) to understand where they are and allow them to experience all of the content in the panel. If we were to place focus on an element inside of the panel, they may not understand there is other information above the current focused element. The detail pane would have a title announced to the user to provide more context. And, when the user is done with the panel and closes it by clicking the close button or pressing escape, the focus should return to the element that originally opened the panel.

## Steps to Reproduce

1. Navigate to https://engineering.cerner.com/terra-ui/components/terra-slide-panel/slide-panel/slide-panel
2. Click on the "Main Toggle Panel" button in the first example 
3. Use the developer console to inspect the active element: `document.activeElement`

## Expected Behavior
Focus should go to the panel after a user selects an option that opens a detail panel. 

## Possible Solution
Look at adding a tabindex to the panel so focus can be shifted to it instead of the first interact-able element within the panel. May need to add an default aria-label to the container with a new prop that allows people to set their own aria-label to "allow the assistive technology user (screen reader for instance) to understand where they are"

Need to test this with VoiceOver on iOS and sometimes it has shifting focus programmatically to non-interactive elements.

## Environment

* Component Name and Version: terra-slide-panel

## @ Mentions
@scottwilmarth

**Comment by nramamurth** [comment]
@bjankord 

Upon opening the Slide Panel, [the focus _does_ shift to the panel element](https://github.com/cerner/terra-framework/blob/master/packages/terra-slide-panel/src/SlidePanel.jsx#L70). The reason that the focus is being shifted to the button in the example stated in the description is because of this logic below that is specific to the example - 

https://github.com/cerner/terra-framework/blob/b38907ec1a9d98cfd3573f15363eba90963c8103/packages/terra-slide-panel/src/terra-dev-site/doc/example/DefaultSlidePanel.jsx#L88-L96

---

This has been fixed as part of https://github.com/cerner/terra-framework/pull/584 and that can be tested on a [test example](https://engineering.cerner.com/terra-framework/tests/terra-slide-panel/slide-panel/slide-panel-toggle).

I don't think this is a bug except for adding a default `aria-label` to the content and probably correcting that example on the site.

**Comment by avinashg1994** [comment]
@scottwilmarth , @neilpfeiffer 
I was wondering what the default `aria-label` value should read if it is not supplied by the consumer.
Should it be "Details Panel" ?

**Comment by neilpfeiffer** [comment]
I don't think it should have a default prop value -- as the generic slide-panel containter can be used for a wide variety of uses well beyond the common Details Panel, include some layouts using the child panel as the "primary"/navigation panel content that launches pre-opened with the `panelBehavior=squish` prop value. 

I would prefer to leave the new `ariaLabel` prop with no default value - force consumers to add it if their situation needs it for accessibity and label it correctly for their usage. 

Also posted on PR here: https://github.com/cerner/terra-framework/pull/900#discussion_r327647080
